src/data/codingcontracttypes.ts

```diff
@@ -794,6 +795,135 @@ export const codingContractTypesMetadata: ICodingContractTypeMetadata[] = [
       return obstacleGrid[obstacleGrid.length - 1][obstacleGrid[0].length - 1] === parseInt(ans);
     },
   },
+  {
+    name: "Shortest Path in a Grid",
+    desc: (data: number[][]): string => {
+      return [
+        "You are located in the top-left corner of the following grid:\n\n",
+        `&nbsp;&nbsp;[${data.map(line => "[" + line + "]").join(",\n&nbsp;&nbsp;&nbsp;")}]\n\n`,
+        "You are trying to find the shortest path to the bottom-right corner of the grid,",
+        "but there are obstacles on the grid that you cannot move onto.",
+        "These obstacles are denoted by '1', while empty spaces are denoted by 0.\n\n",
+        "Determine the shortest path from start to finish, if one exists.",
+        "The answer should be given as a string of UDLR characters, indicating the moves along the path\n\n",
+        "NOTE: If there are multiple equally short paths, any of them is accepted as answer.",
+        "If there is no path, the answer should be an empty string.\n",
+        "NOTE: The data returned for this contract is an 2D array of numbers representing the grid.\n\n",
+        "Examples:\n\n",
+        "&nbsp;&nbsp;&nbsp;&nbsp;[[0,1,0,0,0],\n",
+        "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[0,0,0,1,0]]\n",
+        "\n",
+        "Answer: 'DRRURRD'\n\n",
+        "&nbsp;&nbsp;&nbsp;&nbsp;[[0,1],\n",
+        "&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[1,0]]\n",
+        "\n",
+        "Answer: ''\n\n",
+      ].join(" ");
+    },
+    difficulty: 7,
+    numTries: 10,
+    gen: (): number[][] => {
+      const height = getRandomInt(6, 12);
+      const width = getRandomInt(6, 12);
+      const dstY = height - 1;
+      const dstX = width - 1;
+      const minPathLength = dstY + dstX; // Math.abs(dstY - srcY) + Math.abs(dstX - srcX)
+
+      const grid: number[][] = new Array(height);
+      for(let y = 0; y < height; y++)
+        grid[y] = new Array(width).fill(0);
+
+      for(let y = 0; y < height; y++) {
+        for(let x = 0; x < width; x++) {
+          if(y == 0 && x == 0) continue; // Don't block start
```

Comment:
we could consider allowing the start and end being blockable and just expect "" also, more edges for the dev to handle and come across at a random time

Reply:
The problem starts with the "character" standing at the (0,0) grid location, which requires that location to not be blocked.

Blocking the end could be OK, but if it is blockable then the obstacle generation weights would also need to be adjusted. Just removing the "Don't block destination" special case drops solvable puzzles ratio to ~66%.

Reply:
no worries :)